pulsar-broker/src/test/java/org/apache/pulsar/broker/stats/ConsumerStatsTest.java

```diff
@@ -480,6 +480,10 @@ public void testAvgMessagesPerEntry() throws Exception {
         metadataConsumer.put("matchValueReschedule", "producer2");
         @Cleanup
         Consumer<String> consumer = pulsarClient.newConsumer(Schema.STRING).topic(topic).properties(metadataConsumer)
+                // Since https://github.com/apache/pulsar/pull/23931 improved the performance of delivery, the consumer
+                // will get more messages than before(it only receives 1 messages at the first delivery), we set queue
+                // size to `1` to keep the test passing.
+                .receiverQueueSize(1)
```

Comment:
I'm not sure why set queue to 1 will get the issue passed. But from the purpose of this test method, it looks like it should not related to the queue size. @poorbarcode Could you please provide more context about this change?

Reply:
> Could you please provide more context about this change?

- In original design, the first read of a subscription only has one message, because broker calculate the avg message size by `msg read out / count`, so it read `1` message when `{msg read out}` is `0`
- This test always assumed the quantity of messages read out at the first time is `1`, see: https://github.com/apache/pulsar/pull/23931/files#diff-6e0b3de39cd005ecdedb9375327d65773640082cf70d44407ec79b2d553080b8R508-L505

Reply:
Ok, so how about change the below assert instead of adding the queue size config?

Reply:
I think that make sense to not let the test only test the consumer with 1 queue size.

Reply:
Sure

Reply:
@codelipenghui 

- The main purpose of the test is the mechanism of `AvgMessagesPerEntry`: `Math.round( {pervious avg} * 0.9 + {new avg} * 0.1 )`
- Without a param `queueSize`, we need at least send `1000` messages 😂 
- I changed the `queueSize` to `20` and changed the test